client/serviceregistration/address.go

```diff
@@ -134,7 +134,7 @@ func GetAddress(
 
 		return driverNet.IP, port, nil
 
-	case structs.AddressModeAlloc:
+	case structs.AddressModeAlloc, structs.AddressModeAllocAdvertiseIPv6:
```

Comment:
The existing error messages below all refer to `address_mode = "alloc"` and we're adding a bunch of conditionals here for the IPv6 mode. I think these want to be different cases and maybe there's some helper functions we can factor out.

Reply:
I've added a helper function `getAddressPort`